### DISCUSSION TITLE
How to define a prisma schema model where at least one of 2 fields are required ( but not both )

### DISCUSSION THREAD

**Discussion by slowfound** [prisma/prisma#21214] [Q&A] [answered]
### Question

**Issue Description:**

I am working on setting up a Prisma schema for a user model in my project, and I want to enforce a requirement that a user must provide either an email address or a phone number during registration, but not both.

Thank you for your assistance!

**Comment by ludralph** [comment]
Hi @Accretence 👋 

Thank you for raising his question.

Prisma does not directly support conditional requirements like requiring either an `email` or a `phone number` but not both. However, you can handle this at the application level.
In your Prisma schema, you can define both email and phoneNumber fields as optional:
```prisma
model User {
  id       Int      @id @default(autoincrement())
  email    String?  @unique
  phone    String?  @unique
  // Other user fields
}
```
Then, in your application logic, you can enforce the requirement that a user must provide either an email address or a phone number during registration, but not both. This would involve checking the input data before it's passed to the Prisma client for database operations.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nikolasburk** [reply, marked as answer]
Hey @Accretence, as @ludralph mentions, you can't express such a constraint in the Prisma schema right now. However, you can use SQL `CHECK` constraints to express this requirement (similar to [here](https://dba.stackexchange.com/questions/190505/create-a-constraint-such-that-only-one-of-two-fields-must-be-filled)). You can still use Prisma Migrate for this by [customizing your migration](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#customizing-migrations).
